app/scripts/lib/AccountIdentitiesPetnamesBridge.test.ts

```diff
@@ -129,24 +109,45 @@ function createNameControllerMock(
 }
 
 describe('AccountIdentitiesPetnamesBridge', () => {
+  let messenger: jest.Mocked<
+    PetnamesBridgeMessenger<
+      AccountIdentitiesPetnamesBridgeEvents,
+      AccountIdentitiesPetnamesBridgeActions
+    >
+  >;
+
   beforeEach(() => {
+    messenger = createMessengerMock();
+  });
+
+  afterEach(() => {
     jest.resetAllMocks();
   });
 
   it('adds petnames entry when account id entry added', () => {
-    const preferencesController = createPreferencesControllerMock(
-      EMPTY_PREFERENCES_STATE,
-    );
     const nameController = createNameControllerMock(EMPTY_NAME_STATE);
     const bridge = new AccountIdentitiesPetnamesBridge({
-      preferencesController,
       nameController,
-      messenger: {} as any,
+      messenger,
     });
     bridge.init();
 
-    preferencesController.updateMockStateAndTriggerListener(
-      createPreferencesStateWithIdentity(ADDRESS_MOCK, NAME_MOCK),
+    // mock listAccounts call
+    messenger.call.mockReturnValue([MOCK_INTERNAL_ACCOUNT]);
+
+    const listener = messenger.subscribe.mock.calls[0][1] as (
```

Comment:
Minor, but as it's a bit verbose to include in each test, we'd usually make a helper function such as:

`simulateSubscribe(messenger: jest.Mocked<AccountIdentitiesPetnamesBridgeMessenger>, stateChange: AccountsControllerState, patch: any[])`

Reply:
Sure, that makes sense.